### PR TITLE
Update: fix git_url, add maintainer, bump to latest version

### LINKED ISSUE #2: Incorrect/Out-of-Date Package
@conda-forge/core Hi there! I am the maintainer of this project ever since it was moved from https://github.com/coagulant/coveralls-python to https://github.com/coveralls-clients/coveralls-python

I notice some of the details in this feedstock are incorrect/out-of-date (the repo url has moved, the latest release is 1.2.0 but recorded here as 1.1, etc). I'm very new to conda and I see you do much of this sort of stuff automatically -- how should I/we go about fixing this?

Should I open a PR against `recipe/meta.yaml`? Will I need to do this for each new release?

### PR DESCRIPTION
**Pull request by TheKevJames**
Fixes #2

@ocefpaf please let me know if this looks right! :)

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -34,17 +32,18 @@ test:
...
   license_family: MIT
-  license_file: LICENCE  # Yes it is a typo, will be fixed in the next release
+  license_file: LICENSE.txt
   summary: Show coverage stats online via coveralls.io
   description: |
-    Coveralls.io is service to publish your coverage stats online with a lot
-    of nice features. This package provides seamless integration with
-    coverage.py in your python projects.
-  dev_url: https://github.com/coagulant/coveralls-python
+    coveralls.io is a service for publishing your coverage stats online. This
```
Comment (ocefpaf) [inline recipe/meta.yaml:41]: I noted that you probably copied the info here to the README file.
We only edit the recipe and use `conda-smithy`, a conda-forge tool, to "re-render" the feedstock.

For more info see: https://conda-forge.org/docs/conda_smithy.html

File: recipe/meta.yaml [hunk 2]
```diff
@@ -34,17 +32,18 @@ test:
...
+    package provides seamless integration with coverage.py (and thus py.test,
+    nosetests, etc...) in your Python projects.
+  dev_url: {{ git_url }}
 
 extra:
   recipe-maintainers:
     - goanpeca
+    - thekevjames
```
Comment (ocefpaf) [inline recipe/meta.yaml:49]: :+1:

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,16 +1,14 @@
...
+{% set version = "1.2.0" %}
+{% set git_url = "https://github.com/coveralls-clients/coveralls-python" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  fn: {{ name }}-{{ version }}.tar.gz
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
```
Comment (ocefpaf) [inline recipe/meta.yaml]: We prefer builds from released tarballs, all you have to do is to update the version number and the `sha256` that we get from [here](https://pypi.org/project/coveralls/#files).

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,16 +1,14 @@
...
-  {{ hash_type }}: {{ hash }}
```
Comment (ocefpaf) [inline recipe/meta.yaml]: This `hash_type/hash` jinja2 is silly. You can modify that to be just `sha256: 510682001517bcca1def9f6252df6ce730fcb9831c62d9fff7c7d55b6fdabdf3` (that is already the sha256 for version 1.2.0.)

File: recipe/meta.yaml [hunk 5]
```diff
@@ -1,16 +1,14 @@
...
-{% set hash = "34160385c13b0c43691ab11c080d4b10dabe3280fc0b2173c731efc5db836808" %}
+{% set version = "1.2.0" %}
+{% set git_url = "https://github.com/coveralls-clients/coveralls-python" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  fn: {{ name }}-{{ version }}.tar.gz
```
Comment (ocefpaf) [inline recipe/meta.yaml:11]: The `fn` line can go. Latest version of conda don't need that.

File: recipe/meta.yaml [hunk 6]
```diff
@@ -1,16 +1,14 @@
 {% set name = "coveralls" %}
-{% set version = "1.1" %}
-{% set hash_type = "sha256" %}
-{% set hash = "34160385c13b0c43691ab11c080d4b10dabe3280fc0b2173c731efc5db836808" %}
+{% set version = "1.2.0" %}
+{% set git_url = "https://github.com/coveralls-clients/coveralls-python" %}
```
Comment (ocefpaf) [inline recipe/meta.yaml]: For readability I'd paste the URL in the metadata directly without this variable. (And if you follow my suggestion to use the tarball that variable would be used only once, so no need for a variable :smile:)

File: recipe/meta.yaml [hunk 7]
```diff
@@ -1,16 +1,14 @@
 {% set name = "coveralls" %}
-{% set version = "1.1" %}
-{% set hash_type = "sha256" %}
```
Comment (ocefpaf) [inline recipe/meta.yaml:3]: This can go, see below.

**Review by ocefpaf — COMMENTED** [review]
I made a few comments to help get started with conda-forge. Please ping me if you have any questions.

File: recipe/meta.yaml (hunk 1 above)
Comment (TheKevJames) [reply recipe/meta.yaml:41]: Do I need to run smithy, or is that done automagically?

**Comment by TheKevJames** [comment]
@ocefpaf thanks for the review, updated in response to comments

File: recipe/meta.yaml (hunk 1 above)
Comment (ocefpaf) [reply recipe/meta.yaml:41]: No need to run it here unless you want to see how it works. We have a bot that run it periodically, but we do recommend people to always re-render to give the bot a break sometimes :wink:

**Comment by ocefpaf** [comment]
@TheKevJames with `conda` all the dependencies must be explicitly listed and not downloaded at build time. It seems we are missing `pytest-runner`. Maybe more...

**Comment by TheKevJames** [comment]
Hmm, looks like @goanpeca didn't add the `setup_requires` or `tests_require`. I assume I only need the former?

I also notice that `PyYAML` is included unconditionally, whereas on PyPI it is only included as a part of `coveralls[yaml]`. Is that something we can emulate here? Looks like there was a [conda issue](https://github.com/conda/conda/issues/1696) for this, but I don't think the resolution quite covers that case.

**Comment by ocefpaf** [comment]
> Hmm, looks like @goanpeca didn't add the `setup_requires` or `tests_require`. I assume I only need the former?

In theory yes. You only need the former. I would need to take a look at the `setup.py` to understand why it tries to install that at build time. Also, conda has the concept of test requires, but it would not help here b/c the error happened at build time.

> I also notice that `PyYAML` is included unconditionally, whereas on PyPI it is only included as a part of `coveralls[yaml]`. Is that something we can emulate here? Looks like there was a [conda issue](https://github.com/conda/conda/issues/1696) for this, but I don't think the resolution quite covers that case.

Yep. We cannot emulate optional dependencies with conda yet. The rule of thumb is to add the optional dependencies that people deem useful by, but with parsimony to avoid creating a "heavy" package.

---
Edit: it seems that `pytest-runner` is listed as `setup_requires` requires. See https://github.com/coveralls-clients/coveralls-python/blob/dc8854c9ca2ee77742cb7af0ce4ead83211281a8/setup.py#L30

**Comment by ocefpaf** [comment]
@TheKevJames this looks good to go. Do you want to try to re-render to understand what would happen or are you good with the PR? (I'll merge as soon as AppVeyor finishes.)

**Comment by TheKevJames** [comment]
The `setup.py` is [here](https://github.com/coveralls-clients/coveralls-python/blob/master/setup.py). I believe we use `pytest-runner` for easier integration with our test suite -- I've created [this issue](https://github.com/coveralls-clients/coveralls-python/issues/169) to do away with that dependency, and will update on this feedstock when that's been completed.

I'm of two minds on the `PyYAML` dependency. We're trying to move away from having it and no longer install it by default on PyPI to help create a slimmer package, but many users still require it. I'll leave it for now, but will likely remove it from the 1.3.0 release, which will include a bit better handling of the case where `PyYAML` is not installed.

I'm happy with it as-is. Merge away! And thanks again for all the help :)

**Comment by ocefpaf** [comment]
> I'm of two minds on the `PyYAML` dependency. We're trying to move away from having it and no longer install it by default on PyPI to help create a slimmer package, but many users still require it. I'll leave it for now, but will likely remove it from the 1.3.0 release, which will include a bit better handling of the case where `PyYAML` is not installed.

:+1: 

> I'm happy with it as-is. Merge away! And thanks again for all the help :)

No problem. AppVeyor is quite backlogged due to some timeouts, not sure what is going on. Merging this now and I'll watch the build on `master`.

**Pull request merged**